### PR TITLE
Added an example to apply ASP on existed BERT training scripts

### PR DESCRIPTION
**Pull request by mingxu1067**
### PR types

New features

### PR changes

Models
### Description

Added an example to apply Automatic SParsity (ASP) module (new feature of PaddlePaddle) on BERT GLUE finetuning.
That is only few line insertion needed to enable training in ASP workflow.
1. Setup layers which would like to keep dense.
2. Wrap optimizer by `sparsity.decorate()`, which would insert weight masking operations.
3. Pruning models to desired sparse pattern (default is 2:4 sparse pattern ) before training start.

Please refer to [ASP PR](https://github.com/PaddlePaddle/Paddle/pull/33132) on PaddlePaddle for more details.

Note: We also implemented Distributed API for ASP, and it's under reviewing ([here](https://github.com/PaddlePaddle/Paddle/pull/33558)). After merged, we would also provided an example on multi-devices BERT pretraining.

### CODE REVIEW

**Comment by ZeyuChen** [comment]
Thanks for your contributions! @guoshengCS Please review the code.

File: examples/language_model/bert/static/run_glue_with_sparaity.py [hunk 1]
```diff
@@ -0,0 +1,456 @@
...
+        optimizer = paddle.optimizer.AdamW(
+            learning_rate=lr_scheduler,
+            epsilon=args.adam_epsilon,
+            parameters=model.parameters(),
+            weight_decay=args.weight_decay,
+            apply_decay_param_fun=lambda x: x in decay_params)
+
+        # Keep Pooler and task-specific layer dense.
+        # Please note, excluded_layers must be set before calling `optimizer.minimize()`.
+        sparsity.set_excluded_layers(main_program, ['linear_72', 'linear_73',])
```
Comment (guoshengCS) [inline examples/language_model/bert/static/run_glue_with_sparaity.py]: How about using `model.bert.pooler.dense.full_name(), model.classifier.full_name()`  to replace `'linear_72', 'linear_73'`, which might be more meaningful.

Comment (mingxu1067) [reply examples/language_model/bert/static/run_glue_with_sparaity.py]: It's a great suggestion. Updated.

**Review by guoshengCS — APPROVED** [review]
LGTM

**Comment by ZeyuChen** [comment]
Thanks for your contribution! @mingxu1067
